### PR TITLE
Makefile, build and tests

### PR DESCRIPTION
**Pull request by aravindavk**
- Added Makefile
- Added golint and gofmt tests
- Fixed golint issues in metric_brick.go file
- Added Gopkg and dep based vendor management
- Added centos-ci script to integrate with gluster CI
- Renamed gluster_exporter to gluster-exporter
- Formatted the *.go files using gofmt -s -w *.go

Signed-off-by: Aravinda VK <avishwan@redhat.com>

### CODE REVIEW

File: extras/centos-ci.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# This script will be called by the gluster_glusterd2 job script on centos-ci.
+# This script sets up the centos-ci environment and runs the PR tests for GD2.
```
Comment (Madhu-1) [inline extras/centos-ci.sh]: this one also need to be corrected

File: scripts/pkg-version [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# To override version/release from git,
+# create VERSION file containing text with version/release
+# eg. v3.4.0-1
+[[ -f VERSION ]] && source VERSION
+PKG_VERSION=${PKG_VERSION:-$(git describe --tags --match "v[0-9]*")}
+
+get_version()
+{
+    # tags and output versions:
```
[the remark this answers is not in the document]
Comment (aravindavk) [reply scripts/pkg-version]: commented stuff are examples to show how it will convert the different version formats

**Comment by aravindavk** [comment]
@shtripat @Madhu-1 fixed gd2 references, please review

File: gluster-exporter/main.go [hunk 3]
```diff
@@ -4,25 +4,34 @@ import (
...
+// ExporterVersion and GitSHA
+// These are set as flags during build time. The current values are just placeholders
 var (
-	port = flag.Int("port", 8080, "Exporter Port")
-	metricsPath = flag.String("metrics-path", "/metrics", "Metrics API Path")
-	peerid = flag.String("peerid", "", "Gluster Node's peer ID")
-	volinfo = flag.String("volinfo", "", "Volume info json file")
+	ExporterVersion         = ""
+	GitSHA                  = ""
+	defaultGlusterd1Workdir = ""
```
[the remark this answers is not in the document]
Comment (aravindavk) [reply gluster-exporter/main.go]: will be used to extract local peer id and other information 
https://github.com/gluster/gluster-prometheus/pull/33

File: gluster-exporter/main.go [hunk 4]
```diff
@@ -4,25 +4,34 @@ import (
...
+	defaultGlusterd2Workdir = ""
+)
 
+var (
+	port                          = flag.Int("port", 8080, "Exporter Port")
+	metricsPath                   = flag.String("metrics-path", "/metrics", "Metrics API Path")
+	peerid                        = flag.String("peerid", "", "Gluster Node's peer ID")
+	volinfo                       = flag.String("volinfo", "", "Volume info json file")
```
[the remark this answers is not in the document]
Comment (aravindavk) [reply gluster-exporter/main.go]: volinfo part will be changed with PR https://github.com/gluster/gluster-prometheus/pull/35

and peerID logic will change with PR https://github.com/gluster/gluster-prometheus/pull/35

**Comment by aravindavk** [comment]
Thanks for reviewing quickly.

**Pull request merged**